[2025-04-03T14:06:27.089+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-03T14:06:27.114+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: news_api_collection.fetch_news_articles manual__2025-04-03T13:54:28.177967+00:00 [queued]>
[2025-04-03T14:06:27.130+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: news_api_collection.fetch_news_articles manual__2025-04-03T13:54:28.177967+00:00 [queued]>
[2025-04-03T14:06:27.130+0000] {taskinstance.py:2867} INFO - Starting attempt 2 of 2
[2025-04-03T14:06:27.153+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): fetch_news_articles> on 2025-04-03 13:54:28.177967+00:00
[2025-04-03T14:06:27.181+0000] {standard_task_runner.py:72} INFO - Started process 116198 to run task
[2025-04-03T14:06:27.187+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'news_api_collection', 'fetch_news_articles', 'manual__2025-04-03T13:54:28.177967+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/news_scrapper.py', '--cfg-path', '/tmp/tmpy72_hr02']
[2025-04-03T14:06:27.189+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask fetch_news_articles
[2025-04-03T14:06:27.244+0000] {task_command.py:467} INFO - Running <TaskInstance: news_api_collection.fetch_news_articles manual__2025-04-03T13:54:28.177967+00:00 [running]> on host hamza-HP-ProBook-450-G8-Notebook-PC
[2025-04-03T14:06:27.338+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='retail_mlops' AIRFLOW_CTX_DAG_ID='news_api_collection' AIRFLOW_CTX_TASK_ID='fetch_news_articles' AIRFLOW_CTX_EXECUTION_DATE='2025-04-03T13:54:28.177967+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-03T13:54:28.177967+00:00'
[2025-04-03T14:06:27.340+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-03T14:06:27.341+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-03T14:06:27.341+0000] {logging_mixin.py:190} INFO - Current task name:fetch_news_articles state:running start_date:2025-04-03 14:06:27.115758+00:00
[2025-04-03T14:06:27.341+0000] {logging_mixin.py:190} INFO - Dag name:news_api_collection and current dag run status:running
[2025-04-03T14:06:27.341+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-04-03T14:06:29.698+0000] {document_converter.py:269} INFO - Going to convert document batch...
[2025-04-03T14:06:29.699+0000] {document_converter.py:304} INFO - Initializing pipeline for SimplePipeline with options hash 4cc01982ae99b46a2a63fcda46c47c35
[2025-04-03T14:06:29.699+0000] {base_pipeline.py:38} INFO - Processing document 65-of-amazon-sellers-raised-prices-in-2024-did-shoppers-notice
[2025-04-03T14:06:29.749+0000] {document_converter.py:284} INFO - Finished converting document 65-of-amazon-sellers-raised-prices-in-2024-did-shoppers-notice in 2.18 sec.
[2025-04-03T14:06:31.363+0000] {document_converter.py:269} INFO - Going to convert document batch...
[2025-04-03T14:06:31.364+0000] {document_converter.py:304} INFO - Initializing pipeline for SimplePipeline with options hash 4cc01982ae99b46a2a63fcda46c47c35
[2025-04-03T14:06:31.364+0000] {base_pipeline.py:38} INFO - Processing document counterfeit-android-devices-found-preloaded-with-triada-malware
[2025-04-03T14:06:31.391+0000] {document_converter.py:284} INFO - Finished converting document counterfeit-android-devices-found-preloaded-with-triada-malware in 1.58 sec.
[2025-04-03T14:06:31.422+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-04-03T14:06:31.441+0000] {process_utils.py:132} INFO - Sending 15 to group 116198. PIDs of all processes in the group: [116198]
[2025-04-03T14:06:31.442+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 116198
[2025-04-03T14:06:31.626+0000] {taskinstance.py:3094} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-04-03T14:06:31.635+0000] {taskinstance.py:3095} ERROR - Stacktrace: 
  File "/home/hamza/Documents/project/Mlops/myenv/bin/airflow", line 8, in <module>
    sys.exit(main())
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/__main__.py", line 62, in main
    args.func(args)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 254, in _run_task_by_selected_method
    return _run_task_by_local_task_job(args, ti)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 322, in _run_task_by_local_task_job
    ret = run_job(job=job_runner.job, execute_callable=job_runner._execute)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/jobs/job.py", line 421, in run_job
    return execute_job(job, execute_callable=execute_callable)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/jobs/job.py", line 450, in execute_job
    ret = execute_callable()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/jobs/local_task_job_runner.py", line 171, in _execute
    self.task_runner.start()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 55, in start
    self.process = self._start_by_fork()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/airflow/dags/news_scrapper.py", line 83, in fetch_news_articles
    article['content_md'] = DC.convert(source, max_num_pages=5).document.export_to_markdown()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/pydantic/_internal/_validate_call.py", line 39, in wrapper_function
    return wrapper(*args, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/pydantic/_internal/_validate_call.py", line 136, in __call__
    res = self.__pydantic_validator__.validate_python(pydantic_core.ArgsKwargs(args, kwargs))
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/document_converter.py", line 220, in convert
    return next(all_res)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/document_converter.py", line 243, in convert_all
    for conv_res in conv_res_iter:
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/document_converter.py", line 265, in _convert
    for input_batch in chunkify(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/utils/utils.py", line 15, in chunkify
    for first in iterator:  # Take the first element from the iterator
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/datamodel/document.py", line 241, in docs
    resolve_source_to_stream(item, self.headers)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling_core/utils/file.py", line 80, in resolve_source_to_stream
    res = requests.get(http_url, stream=True, headers=req_headers)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    response = self._make_request(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 464, in _make_request
    self._validate_conn(conn)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 1093, in _validate_conn
    conn.connect()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connection.py", line 704, in connect
    self.sock = sock = self._new_conn()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connection.py", line 198, in _new_conn
    sock = connection.create_connection(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/util/connection.py", line 60, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.12/socket.py", line 963, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3095, in signal_handler
    self.log.error("Stacktrace: \n%s", "".join(traceback.format_stack()))

[2025-04-03T14:06:31.644+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/airflow/dags/news_scrapper.py", line 83, in fetch_news_articles
    article['content_md'] = DC.convert(source, max_num_pages=5).document.export_to_markdown()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/pydantic/_internal/_validate_call.py", line 39, in wrapper_function
    return wrapper(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/pydantic/_internal/_validate_call.py", line 136, in __call__
    res = self.__pydantic_validator__.validate_python(pydantic_core.ArgsKwargs(args, kwargs))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/document_converter.py", line 220, in convert
    return next(all_res)
           ^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/document_converter.py", line 243, in convert_all
    for conv_res in conv_res_iter:
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/document_converter.py", line 265, in _convert
    for input_batch in chunkify(
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/utils/utils.py", line 15, in chunkify
    for first in iterator:  # Take the first element from the iterator
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling/datamodel/document.py", line 241, in docs
    resolve_source_to_stream(item, self.headers)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/docling_core/utils/file.py", line 80, in resolve_source_to_stream
    res = requests.get(http_url, stream=True, headers=req_headers)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 464, in _make_request
    self._validate_conn(conn)
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 1093, in _validate_conn
    conn.connect()
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connection.py", line 704, in connect
    self.sock = sock = self._new_conn()
                       ^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/connection.py", line 198, in _new_conn
    sock = connection.create_connection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/urllib3/util/connection.py", line 60, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/socket.py", line 963, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/hamza/Documents/project/Mlops/myenv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3097, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2025-04-03T14:06:31.650+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-04-03T14:06:31.650+0000] {logging_mixin.py:190} INFO - Task start:2025-04-03 14:06:27.115758+00:00 end:2025-04-03 14:06:31.649591+00:00 duration:4.533833
[2025-04-03T14:06:31.651+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): fetch_news_articles> dag:<DAG: news_api_collection> dagrun:<DagRun news_api_collection @ 2025-04-03 13:54:28.177967+00:00: manual__2025-04-03T13:54:28.177967+00:00, state:running, queued_at: 2025-04-03 13:54:28.233592+00:00. externally triggered: True>
[2025-04-03T14:06:31.651+0000] {logging_mixin.py:190} INFO - Failure caused by Task received SIGTERM signal
[2025-04-03T14:06:31.651+0000] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=news_api_collection, task_id=fetch_news_articles, run_id=manual__2025-04-03T13:54:28.177967+00:00, execution_date=20250403T135428, start_date=20250403T140627, end_date=20250403T140631
[2025-04-03T14:06:31.680+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-04-03T14:06:31.735+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=116198, status='terminated', exitcode=2, started='14:06:26') (116198) terminated with exit code 2
